src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,13 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    /**
+     * Check if the command mutates data in the address book
+     *
+     * @return true if the command mutates data in the address book
+     */
+    public boolean isMutating() {
+        throw new UnsupportedOperationException("This method should be implement in child classes");
```

Comment:
Consider making this method abstract instead. This forces the child classes to implement the method, instead of relying on runtime exceptions.